### PR TITLE
etcdctl: tighten up output, reorganize README.md

### PR DESCRIPTION
**Pull request by heyitsanthony**
Documentation was far too repetitive, making it a chore to read and
make changes. All commands are now organized by functionality and all
repetitive bits about return values and output are in a generalized
subsections.

etcdctl's output handling was missing a lot of commands. Similarly,
in many cases an output format could be given but fail to report
an error as expected.

### CODE REVIEW

File: etcdctl/ctlv3/command/printer_protobuf.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	m, ok := v.(pbMarshal)
+	if !ok {
+		ExitWithError(ExitBadFeature, fmt.Errorf("marshal unsupported for %v", m))
+	}
+	b, err := m.Marshal()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "%v\n", err)
+		return
+	}
+	fmt.Printf(string(b))
```
Comment (gyuho) [inline etcdctl/ctlv3/command/printer_protobuf.go]: Q. This doesn't have `\n` at the end. Is that what we want?

File: etcdctl/ctlv3/command/printer_fields.go [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+		default:
+			fmt.Printf("\"Unknown\" : %q\n", fmt.Sprintf("%+v", v))
+		}
+	}
+}
+
+func (p *fieldsPrinter) Watch(resp v3.WatchResponse) {
+	p.hdr(&resp.Header)
+	for _, e := range resp.Events {
+		fmt.Println(`"Type" : `, e.Type)
```
Comment (gyuho) [inline etcdctl/ctlv3/command/printer_fields.go]: `fmt.Println` will insert another whitespace? Maybe just `fmt.Println(`"Type" :`, e.Type)`?

File: etcdctl/ctlv3/command/printer_protobuf.go (hunk 1 above)
Comment (heyitsanthony) [inline etcdctl/ctlv3/command/printer_protobuf.go]: yeah, it needs to concatenate; there wouldn't be a newline in a protobuf stream over the network

File: etcdctl/ctlv3/command/printer_fields.go (hunk 2 above)
Comment (heyitsanthony) [inline etcdctl/ctlv3/command/printer_fields.go]: sure

File: etcdctl/ctlv3/command/printer_protobuf.go (hunk 1 above)
Comment (gyuho) [inline etcdctl/ctlv3/command/printer_protobuf.go]: Ok makes sense.

**Comment by gyuho** [comment]
lgtm. /cc @xiang90

**Comment by xiang90** [comment]
lgtm

**Pull request merged**